pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1593,38 +1626,27 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 				return err
 			}
 			inputDevices = append(inputDevices, inputDevice)
-			if inputDevice.Bus == "usb" {
-				isUSBDevicePresent = true
-			}
 		}
 		domain.Spec.Devices.Inputs = inputDevices
 	}
 
-	isUSBRedirEnabled, err := Convert_v1_Usbredir_To_api_Usbredir(vmi, &domain.Spec.Devices, c)
+	err = Convert_v1_Usbredir_To_api_Usbredir(vmi, &domain.Spec.Devices, c)
 	if err != nil {
 		return err
 	}
 
-	domain.Spec.Devices.Ballooning = &api.MemBalloon{}
-	ConvertV1ToAPIBalloning(&vmi.Spec.Domain.Devices, domain.Spec.Devices.Ballooning, c)
-
-	//usb controller is turned on, only when user specify input device with usb bus,
-	//otherwise it is turned off
-	//In ppc64le usb devices like mouse / keyboard are set by default,
```

Comment:
This comment seems to assume that the only USB devices are input devices, which is not true anymore.
It's also hard to read, probably worth reworking...

Reply:
I just removed the first 2 lines, thanks for noticing that.